# Graded assignment 1

Mathieu Bettonviel #2014565#


# Packages

We need the package `tidyverse`

In [5]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [10]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3        V4         V5        V6        V7       
1 0.05521273 0.5554052 0.9243167 0.07355309 0.9744580 0.6187735 0.4739643
2 0.55441409 0.6500207 0.7020048 0.19623165 0.3404972 0.6032025 0.1765205
  V8        V9        V10       V11       V12       V13       V14      
1 0.5440232 0.9465162 0.4235213 0.6782474 0.7320044 0.4336928 0.5145764
2 0.5874930 0.4250637 0.8129517 0.8026975 0.7383455 0.5824650 0.1875700
  V15       V16       V17       V18        V19       V20      
1 0.9336920 0.6937232 0.1965586 0.04800706 0.8321606 0.3546383
2 0.7948546 0.2631860 0.8252308 0.52836801 0.3619848 0.5610841

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [11]:
df2 <- df1 %>%
gather(Key, Value)
  

head(df2)

Key Value     
1 V1  0.05521273
2 V1  0.55441409
3 V2  0.55540516
4 V2  0.65002069
5 V3  0.92431669
6 V3  0.70200482

## Assignment 2

We will generate some data:

In [13]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [14]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [15]:
df5 <- full_join(df3,df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook